src/Pages/AccountPage/SignUp.js

```diff
@@ -0,0 +1,224 @@
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+
+export default function SignUp({ changeType, changeModalOpen }) {
+  const [dlList, setDlList] = useState([]);
+  const [circleClickArr, setcircleClickArr] = useState([]);
+  const [isCircleAll, setIsCircleAll] = useState(false);
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  useEffect(() => {
+    setIsCircleAll(
+      circleClickArr.length > 0 && circleClickArr.every(isClicked => isClicked)
+    );
+  }, [circleClickArr]);
+
+  const fetchData = () => {
+    fetch('/data/SignUp.json')
+      .then(res => res.json())
+      .then(data => {
+        setDlList(data);
+        setcircleClickArr(Array.from({ length: data.length }, () => false));
+      });
+  };
+
+  const selectCircleClick = index => {
+    setcircleClickArr(
+      circleClickArr.map((circle, i) => (i === index ? !circle : circle))
+    );
+  };
+
+  const isAllCircleClick = () => {
+    setIsCircleAll(!isCircleAll);
+    setcircleClickArr(
+      circleClickArr.map(circle => (isCircleAll ? !circle : true))
+    );
+  };
+
+  return (
+    <Signup>
+      <Header>
+        <span>약관동의</span>
+        <XIcon
+          onClick={() => changeModalOpen(false)}
+          className="fas fa-times"
+        />
+      </Header>
+      <Main>
+        {dlList.length > 0 &&
+          dlList.map((dl, index) => {
```

Comment:
```suggestion
        {dlList.map((dl, index) => {
```
초기 값이 배열이므로 에러 안날거고, 어차피 빈배열이라면 아무것도 안나올 것 같습니다.

Reply:
감사합니다 . 다음부턴 PR전에 고치고 싶네요..